Userland/Libraries/LibJS/Runtime/ECMAScriptFunctionObject.cpp

```diff
@@ -699,7 +697,10 @@ void ECMAScriptFunctionObject::ordinary_call_bind_this(ExecutionContext& callee_
     // 7. Assert: localEnv is a function Environment Record.
     // 8. Assert: The next step never returns an abrupt completion because localEnv.[[ThisBindingStatus]] is not initialized.
     // 9. Perform ! localEnv.BindThisValue(thisValue).
-    MUST(verify_cast<FunctionEnvironment>(*local_env).bind_this_value(vm, this_value));
+    if (m_function_environment_needed)
+        MUST(verify_cast<FunctionEnvironment>(*local_env).bind_this_value(vm, this_value));
+    else
+        callee_context.this_value = this_value;
```

Comment:
Perhaps we should always set `callee_context.this_value` here?
Seems nice to keep it consistent, instead of having it populated sometimes, empty sometimes.